giskard/llm/loaders/prompt_injections.py

```diff
@@ -0,0 +1,100 @@
+import ast
+import requests
+from typing import Optional
+import pandas as pd
+
+from giskard.datasets.base import Dataset
+
+INJECTION_DATA_URL = "https://raw.githubusercontent.com/Giskard-AI/prompt-injections/main/prompt_injections.csv"
+GISKARD_META_URL = "https://raw.githubusercontent.com/Giskard-AI/prompt-injections/main/giskard_meta_data.csv"
+
+
+def _check_url(url: str):
+    response = requests.get(url)
+    if response.status_code != 200:
+        raise ConnectionError(f"A problem occured while trying to connect to {url}")
+
+
+def _check_matching_dfs_len(df1, df2):
+    if len(df1) != len(df2):
+        raise ValueError(
+            f"{__name__}: {INJECTION_DATA_URL} and {GISKARD_META_URL} should "
+            "have the same length and should be a one-to-one mapping of each other."
+        )
+
+
+def _check_meta_df_requirements(df):
+    if "substrings" not in df.columns:
+        raise ValueError(f"{__name__}: substrings are needed for the evaluation.")
+
+    if df.substrings.isnull().values.any():
+        raise ValueError(f"{__name__}: substrings column cannot have any NaN values.")
+    df.substrings = df.substrings.apply(ast.literal_eval)
```

Comment:
In my opinion there is no need for all this validation here. This should be checked in the prompt-injection repo.
If there’s a connection error, pandas will raise it.

Reply:
removed the url check. left the others.

Reply:
done